tests/test-present.cpp

```diff
@@ -1395,4 +1396,109 @@ namespace libsemigroups {
     REQUIRE(presentation::longest_common_subword(p) == "");
   }
 
+  LIBSEMIGROUPS_TEST_CASE("word_functions",
+                          "043",
+                          "operator<<",
+                          "[quick][word_functions]") {
+    using namespace literals;
+    using presentation::operator<<;
+    word_type           w = {0, 1};
+    word_type           v = {2};
+    REQUIRE((w << v) == word_type({0, 1, 2}));
+    REQUIRE((w << v << w) == word_type({0, 1, 2, 0, 1}));
+
+    REQUIRE((word_type({0, 1, 0}) << word_type({2}))
+            == word_type({0, 1, 0, 2}));
+    REQUIRE((word_type({0}) << word_type({})) == word_type({0}));
+    REQUIRE((word_type({}) << word_type({0})) == word_type({0}));
+
+    std::string x = "ab";
+    std::string y = "c";
+    REQUIRE((x << y) == "abc");
+    REQUIRE((std::string("") << "a") == "a");
```

Comment:
Be good to have a version of `<<` for `char const* and `char const*` so that this could just be `"" << "a"`. Now that I'm writing this, maybe `<<` should rather be `+` because then we could just do `"" + "a"` and there'd be nothing to implement at all for strings (only for `word_type`), which would also remove the template code, and generally simplify this.